### ISSUE TITLE
Define and build normalised stream of "updates"

### ISSUE THREAD

**Issue by Pinpickle** [hackersatcambridge/hac-website#35]
Our home page features an update stream. It has upcoming events as well as things we've done (videos, blog posts, all the good stuff).

We need a way to combine these two concepts in a way that doesn't result in one being dwarfed by the other (events are in the future, released things are in the past).

Events can be pulled in by #23 but other updates need some kind of representation in our database.

**Comment by CharlieCrisp** [comment]
My thoughts on the matter:
Stuff that we have done in the past will be fairly easy to filter on a newest to oldest basis.

In terms of events, we need a way of filtering which events are relevant at any given time. Provided this filtering works well, we can prioritise the relevant events without swamping the past projects. This assumes that we will never have more than roughly 2-3 relevant events at once, which I think is a fair assumption for about 90% of the time.

Unfortunately, filtering is not so easy because some events will be important even when they are a few months away (e.g. ChariTech) whereas workshops are only relevant in the week before.

I suggest that we perhaps define a notion of a warning period and TTL for events. I.e. two figures we input for each event which describe how far before and after the event they should be advertised.
We can then calculate if we are in the 'live period' for each workshop and use this info to choose whether we display it or not.

Just a suggestion but would be interested to hear thoughts about this :)

**Comment by Pinpickle** [comment]
@CharlieCrisp that looks like it will give us a good amount of freedom going forwards

While this solves the when to display issue, it doesn't solve the sorting issue. How do we intersperse future things with past things?

**Comment by CharlieCrisp** [comment]
@Pinpickle So my thinking was that we show the future things first and then fill the remainder with past things. If we showed 4 things in total then this may mean we get swamped with future events but only when we're very busy. Alternatively we could show like 8 events and then we'd definitely have a good mix.

**Comment by moosichu** [comment]
Will we have dedicated page to view an unlimited number of these "updates" filtered in an arbitrary fashion?

**Comment by Pinpickle** [comment]
@moosichu that is probably useful, but I don't think is necessary for MVP.

**Comment by moosichu** [comment]
Can this issue be closed now?

**Comment by Pinpickle** [comment]
@moosichu I don't believe so - we have figured out events but not really any of the other things.

**Comment by moosichu** [comment]
This issue depends on a load of others, can someone please try and find them all and list them here?

**Comment by jaredkhan** [comment]
Tasks within this include: 
- Designing the data structure of an 'Update'
- Creating a controller that takes many stream providers and outputs a stream of updates
- Creating stream providers for:
   - Workshops
   - Events
   - YouTube videos
   - Medium posts
